### DISCUSSION TITLE
Can I not expose my cron paths to the public?

### DISCUSSION THREAD

**Discussion by gkorland** [vercel/community#4322] [Help]
### Summary

How do I prevent my cron paths from being exposed to the public and can only be used by the cron job?

### Steps to Reproduce

Created a cron job from the example, tried to call it with `curl` and it replied.

**Comment by amyegan** [comment]
Hey @gkorland. There is a special environment variable that is being added to the docs, but it's new and isn't there yet. You can secure your cron jobs by providing an environment variable with the name `CRON_SECRET`. The value will be part of the `Authorization` header of the incoming request so your endpoint can check the value to verify the request's origin.

The process is similar to the current examples of [how to secure cron jobs](https://vercel.com/docs/cron-jobs#how-to-secure-cron-jobs) in the Cron Jobs documentation. The main difference is a `sharedKey` query param versus a `CRON_SECRET` environment variable.

**Comment by gkorland** [comment]
like this?

```
  if (request.headers.get('Authorization') !== process.env.CRON_SECRET) {
    return NextResponse.json({ message: "Unauthorized" }, { status: 401 })
  }
```
